quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/resteasy/TransactionalSessionHandler.java

```diff
@@ -17,46 +17,48 @@
 
 package org.keycloak.quarkus.runtime.integration.resteasy;
 
-import static org.keycloak.common.util.Resteasy.clearContextData;
+import io.quarkus.arc.Arc;
+import io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext;
 
-import jakarta.ws.rs.container.CompletionCallback;
 import org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext;
 import org.jboss.resteasy.reactive.server.spi.ServerRestHandler;
 import org.keycloak.common.util.Resteasy;
 import org.keycloak.models.KeycloakSession;
-import org.keycloak.models.KeycloakSessionFactory;
-import org.keycloak.quarkus.runtime.integration.QuarkusKeycloakSessionFactory;
-import org.keycloak.quarkus.runtime.transaction.TransactionalSessionHandler;
 
-import io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext;
-import io.vertx.ext.web.RoutingContext;
+import static org.keycloak.common.util.Resteasy.clearContextData;
 
-public final class CreateSessionHandler implements ServerRestHandler, TransactionalSessionHandler, CompletionCallback {
+public final class TransactionalSessionHandler implements ServerRestHandler {
```

Comment:
The reason why we introduced this `TransactionalSessionHandler` interface is because we have been supporting different JAX-RS stacks over time and it helps to quickly identify the components that are related to managing the session lifecycle.

IMO, we can just keep it as it has more pros than cons.

Reply:
The rationale for removal here is that this class has the same name as the interface and there's no other implementation using these methods. 

I can of course pull it back out again, but the interface is effectively a marker, but not looked for anywhere, and the methods essentially static.

Reply:
Exactly, that is the reason behind this marker interface.

As we have migrated from Undertow/Widfly it was kinda not clear all the components related to the request/transaction lifecycle when processing requests so we add this interface to mark such components and make more clear what they are.

Reply:
My misunderstanding is probably that it seems specific to just this transaction lifecycle - the other lifecycles - undertow, and KeycloakModelUtils.runJobInTransaction use their own, but now closely aligned logic. It seems like you still need to know to look for KeycloakSessionFactory.create in multiple places, the transaction method methods in multiple places, and the session close in multiple places.

Reply:
KeycloakModelUtils is a separate thing and not really handling the request lifecycle. That is the difference.

Is it so bad if we keep this interface?

Reply:
> Is it so bad if we keep this interface?

I've added a commit of what it would look like with the additional interface.

Reply:
On top of that there's now a commit of what this would look like with using the cdi Disposes annotation.